lib/services/services.c

```diff
@@ -259,7 +259,7 @@ resources_action_create(const char *name, const char *standard,
         free(dirs);
         free(buf);
 
-        if (dir) {
+        if (!pcmk__str_empty(dir)) {
             op->opaque->exec = crm_strdup_printf("%s/%s/%s",
```

Comment:
Ahhh, I get it now: dir is just a pointer into dirs, and you've freed dirs by this point. Keeping buf, and checking buf != NULL instead of pcmk__str_empty(dir) here, should do the trick